src/jit/compiler.cpp

```diff
@@ -4930,6 +4930,14 @@ int           Compiler::compCompileHelper (CORINFO_MODULE_HANDLE            clas
         }
         info.compRetNativeType = info.compRetType         = JITtype2varType(methodInfo->args.retType);
 
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+        // Set the strcutDesc for the return type if struct.
+        if (varTypeIsStruct(info.compRetType))
+        {
+            eeGetSystemVAmd64PassStructInRegisterDescriptor(info.compMethodInfo->args.retTypeClass, &(info.retStructDesc));
+        }
+#endif // FEATURE_UNIX_AMD64_STRUCT_PASSING
```

Comment:
If the only use of this structDesc is in emitting GS Cookie check for a GT_RETURN node, can't this be computing on demand using eeGetSystemVAmd64PassStructInRegisterDescriptor(info.compMethodInfo->args.retTypeClass, &structDesc)?

IMO we are unnecessarily paying the cost of computing structDesc and storing it without knowing whether the method would need it.